### PR TITLE
Fix/component lifecycle issue

### PR DESCRIPTION
**Pull request by rfreitas**
This a fix for the bug #54.

It only solves the lifecycle issues for `Camera` and `Capture`.

The issue is due to `preact` not handling nested pure components well. Sibling and stateful parents often don't get lifecycle method calls if they have pure components associated with them.

The fix found was _impurify_ some components.

This is a hack/workaround and more general fix should be found.

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://71.pr.onfido-sdk-ui.onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://71.pr.onfido-sdk-ui.onfido.surge.sh

**Comment by rfreitas** [comment]
yeah, it goes missing, that UX still needs some tweaking..

**Comment by rfreitas** [comment]
There is still a bug when the document is retaken a second time, the camera unmount method will not be called.

File: src/components/Camera/index.js [hunk 1]
```diff
@@ -100,12 +107,7 @@ export default class Camera extends Component {
       method, onUserMedia, onUploadFallback,
       faceCaptureClick: this.capture.once,
       countDownRef: (c) => { this.countdown = c },
-      webcamRef: (c) => {
```
Comment (rfreitas) [inline src/components/Camera/index.js]: this was the hack to figure out when mounting and unmounting the webcam

File: src/components/Uploader/index.js [hunk 2]
```diff
@@ -38,7 +39,10 @@ export const UploadError = ({errorMessage}) => (
   <div className={`${style.text} ${style.error}`}>{errorMessage}</div>
 )
 
-export const Uploader = ({method, onImageSelected, uploading, noDocument}) => (
+//TODO move to react instead of preact, since preact has issues handling pure components
+//IF this component is exported as pure,
+//some components like Camera will not have componentWillUnmount called
```
Comment (stephencookdev) [inline src/components/Uploader/index.js]: I generally find that when putting in comments like these, it's helpful to add a link to the issue tracking the problem

That way you can have more detail, have an issue for tracking, and also have something to grep for easily when the issue is resolved

**Pull request merged**